jetbrains-core/src/software/aws/toolkits/jetbrains/core/AwsResourceCache.kt

```diff
@@ -199,28 +208,56 @@ class DefaultAwsResourceCache(private val project: Project, private val clock: C
         forceFetch: Boolean
     ): CompletionStage<T> = when (resource) {
         is Resource.View<*, T> -> getResource(resource.underlying, region, credentialProvider, useStale, forceFetch).thenApply { resource.doMap(it as Any) }
-        is Resource.Cached<T> -> Context(resource, region, credentialProvider, useStale, forceFetch).also { getCachedResource(it, 0) }.future
+        is Resource.Cached<T> -> Context(resource, region, credentialProvider, useStale, forceFetch).also { getCachedResource(it) }.future
     }
 
-    private fun <T> getCachedResource(context: Context<T>, failedAttempts: Int) {
+    private fun <T> getCachedResource(context: Context<T>) {
         ApplicationManager.getApplication().executeOnPooledThread {
             try {
                 @Suppress("UNCHECKED_CAST")
                 val result = cache.compute(context.cacheKey) { _, value ->
                     fetchIfNeeded(context, value as Entry<T>?)
                 } as Entry<T>
                 context.future.complete(result.value)
-            } catch (e: AssertionError) { // HACK: until we figure out why Guava Cache is throwing an exception : https://github.com/google/guava/issues/3464
-                when {
-                    failedAttempts > 0 -> context.future.completeExceptionally(e)
-                    else -> getCachedResource(context, failedAttempts + 1)
-                }
             } catch (e: Throwable) {
                 context.future.completeExceptionally(e)
             }
         }
     }
 
+    private fun runCacheMaintenance() {
+        try {
+            val entries = cache.entries.toList()
```

Comment:
Does it make sense to go entries to sequence do the sum onEach and then toList?

Reply:
yeah that probably is more efficient, we'd only be traversing the list once. more mutation - *sigh*